internal/types/subtypes/registry_global.go

```diff
@@ -7,23 +7,29 @@ import (
 	"sync"
 
 	"github.com/hashicorp/boundary/internal/errors"
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/reflect/protoreflect"
 )
 
 var globalRegistry = newRegistry()
 
 // newGlobalRegistry creates a new boundary resource subtype registry.
 func newRegistry() *registry {
 	return &registry{
-		subtypesPrefixes: make(map[string]map[string]Subtype),
-		knownSubtypes:    make(map[string]map[Subtype]interface{}),
+		subtypesPrefixes:            make(map[string]map[string]Subtype),
+		knownSubtypes:               make(map[string]map[Subtype]interface{}),
+		requestTransformationFuncs:  make(map[protoreflect.FullName]TransformationFunc),
+		responseTransformationFuncs: make(map[protoreflect.FullName]TransformationFunc),
 	}
 }
 
 // registry stores a collection of boundary resource subtypes along with their
 // prefixes and allows for translating prefixes back to registered subtypes.
 type registry struct {
-	subtypesPrefixes map[string]map[string]Subtype
-	knownSubtypes    map[string]map[Subtype]interface{}
+	subtypesPrefixes            map[string]map[string]Subtype
+	knownSubtypes               map[string]map[Subtype]interface{}
+	requestTransformationFuncs  map[protoreflect.FullName]TransformationFunc
+	responseTransformationFuncs map[protoreflect.FullName]TransformationFunc
```

Comment:
I am not sure this belongs in this registry, given that the keys, and the input for the corresponding register function is a protomessage, while this registry is currently responsible for mapping Subtypes to ids.

I think it should either be a separate registry, or potentially added into the [attributeRegistery](https://github.com/hashicorp/boundary/blob/a6c026fb769004f1b9beca371ff46d9ed2d53cc2/internal/types/subtypes/attributes.go#L35-L40) or [sourceRegistry](https://github.com/hashicorp/boundary/blob/a6c026fb769004f1b9beca371ff46d9ed2d53cc2/internal/types/subtypes/source_registry.go#L28-L36), but they would probably need to be renamed to be more clear what they are.

Reply:
Good points, let me think a bit about it and update it.

Reply:
Created a new `transformationRegistry` type, let me know if you have another idea.